internal/cluster/connection_manager.go

```diff
@@ -520,89 +524,106 @@ func (m *ConnectionManager) detectFixBrokenConnections() {
 }
 
 type connectionMap struct {
-	connectionsMu *sync.RWMutex
+	lb pubcluster.LoadBalancer
+	mu *sync.RWMutex
 	// addrToConn maps connection address to connection
 	addrToConn map[pubcluster.Address]*Connection
 	// connToAddr maps connection ID to address
 	connToAddr map[int64]pubcluster.Address
+	addrs      []pubcluster.Address
 }
 
-func newConnectionMap() *connectionMap {
+func newConnectionMap(lb pubcluster.LoadBalancer) *connectionMap {
 	return &connectionMap{
-		connectionsMu: &sync.RWMutex{},
-		addrToConn:    map[pubcluster.Address]*Connection{},
-		connToAddr:    map[int64]pubcluster.Address{},
+		lb:         lb,
+		mu:         &sync.RWMutex{},
+		addrToConn: map[pubcluster.Address]*Connection{},
+		connToAddr: map[int64]pubcluster.Address{},
 	}
 }
 
 func (m *connectionMap) AddConnection(conn *Connection, addr pubcluster.Address) {
-	m.connectionsMu.Lock()
+	m.mu.Lock()
+	defer m.mu.Unlock()
+	// if the connection was already added, skip it
+	if _, ok := m.connToAddr[conn.connectionID]; ok {
+		return
+	}
 	m.addrToConn[addr] = conn
 	m.connToAddr[conn.connectionID] = addr
-	m.connectionsMu.Unlock()
+	m.addrs = append(m.addrs, addr)
 }
 
 // RemoveConnection removes a connection and returns true if there are no more addrToConn left.
```

Comment:
I think `returns true if there are no more addrToConn left.` is leftover. Should be `returns number of connections left`

Reply:
Updated at: https://github.com/hazelcast/hazelcast-go-client/pull/580/commits/8ff2f4a9dc5ac6228040a9b24a7045c40fa06bcd